packages/cryptoassets/src/chains/BaseChain.ts

```diff
@@ -0,0 +1,57 @@
+import { ensure0x } from '@chainify/utils';
+import { getAddress, isAddress } from '@ethersproject/address';
+import { IAsset } from '../interfaces/IAsset';
+import { IChain } from '../interfaces/IChain';
+import { IFees } from '../interfaces/IFees';
+import { IGasLimits } from '../interfaces/IGasLimit';
+import { INetwork } from '../interfaces/INetwork';
+import { ChainId, ExplorerView, FeeMultiplier } from '../types';
+
+export abstract class BaseChain implements IChain {
+  constructor(chain: IChain) {
+    Object.assign(this, chain);
+  }
+
+  public id: ChainId;
```

Comment:
I think you can avoid defining all the properties in the class like this:

```
export interface BaseChain implements IChain

export abstract class BaseChain implements IChain {
...
```
Don't ask me how this works 😄